Draft of modslot:any and modslot:none filters.

johyphenel opened the pull request:
Apologies in advance, I don't use github much, new to node and typescript, new to DIM. Please take a look at this first draft and give feedback on:
- Do you like the feature?
- Are there better ways to do any of the changes? 
- Am I missing anything to make this change complete? 

I also noticed that the autoformatter that I think was installed by DIM project in VS code seemed to be making a _lot_ of changes to code I didn't edit... Adding newlines, changing indent, adding spaces after `<var>[ ]?.<property>[ ]?.<property>`. Is there a config or setup step I'm missing to stop this? That said, it seems like some of them were reverted as part of the submit process..? So definitely confused.

johyphenel commented:
Not sure the right way to "officially" reference bugs? But this relates to #4786.

delphiactual commented on src/data/d2/seasonal-mod-slots.json:
```diff
@@ -1,6 +1,8 @@
-  "undying": [1081029832]
+  "undying": [1081029832],
+  "any": [1081029832, 1202876185, 13646368, 1962317640, 2149155760, 2712224971, 65589297]
 }
```
You cannot modify this file, you must submit a PR to https://github.com/DestinyItemManager/d2-additional-info to change the way this file is generated.

delphiactual replied:
https://github.com/DestinyItemManager/d2-additional-info/pull/84

johyphenel replied:
Thanks for the info, and for drafting the PR :) 

Should we just wait on this PR until DestinyItemManager/d2-additional-info#84 is merged then?

delphiactual replied:
merging now

delphiactual commented on src/app/search/search-filters.ts:
```diff
@@ -1018,11 +1018,20 @@ function searchFilters(
       },
       modslot(item: DimItem, predicate: string) {
-        const modSocketHashes = seasonalSocketHashesByName[predicate];
+        if (!item.isDestiny2() || !item.sockets || !(item.bucket?.sort === 'Armor')) return false;
```
```suggestion
        if (!item.isDestiny2() || !item.sockets || !(item.bucket?.sort === 'Armor')) {
          return false;
        }
```

johyphenel replied:
Sorry :( fixed.
My work environment auto runs formatters and linters when you create changes, so I have some bad habits :shame-cube:.

delphiactual commented:
It seems like you didn't fully follow https://github.com/DestinyItemManager/DIM/blob/master/docs/CONTRIBUTING.md as you were able to submit a PR with bad linting meaning something did not get installed correctly. Are you in our Discord by chance, then we can walk through what is missing for you... most likely you missed `yarn install`

delphiactual commented on src/app/search/search-filters.ts:
```diff
@@ -1018,11 +1018,20 @@ function searchFilters(
-            item.isDestiny2() &&
-            item.sockets &&
             item.sockets.sockets.find((socket) =>
               modSocketHashes.includes(socket?.plug?.plugItem?.plug?.plugCategoryHash)
```
dedupe code by doing  the following
```
        if (!item.isDestiny2() || !item.sockets || !(item.bucket?.sort === 'Armor')) {
          return false;
        }
        let isNone = false;
        if (predicate === 'none') {
          isNone = true;
          predicate = 'any';
        }
        const modSocketHashes = seasonalSocketHashesByName[predicate];
        const matches = Boolean(
          item.sockets.sockets.find((socket) =>
            modSocketHashes?.includes(socket?.plug?.plugItem?.plug?.plugCategoryHash)
          )
        );
        return !isNone ? matches : !matches;
```

johyphenel replied:
Good suggestion, thanks! :)

johyphenel commented:
Updated per comments.
- Reverted manual changes to auto-generated config file. 
- Updated modslot code per suggestion.
- Resorted help file to match autogenerated order in DestinyItemManager/d2-additional-info#84 
- Actually ran `yarn lint` this time :o

sundevour commented:
works fab for me. thank you!
